File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length * grid.length;
        int[] count = new int[n + 1]; 
        int[] result = new int[2]; 

        // Count occurrences of each number in the grid
        for (int[] row : grid) {
            for (int num : row) {
                count[num]++;
            }
        }

        // Find the repeated and missing numbers
        for (int i = 1; i <= n; i++) {
            if (count[i] == 2) {
                result[0] = i; 
            } else if (count[i] == 0) {
                result[1] = i; 
            }
        }

        return result;
    }
}

File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.kt
class Solution {
    fun findMissingAndRepeatedValues(grid: Array<IntArray>): IntArray {
        val n = grid.size * grid.size
        val result = IntArray(2)
        val count = IntArray(n + 1)
        
        for(row in grid){
            for(num in row){
                count[num]++
            }
        }

        for(i in 1..n){
            when(count[i]){
                2 -> result[0] = i 
                0 -> result[1] = i
            }
        }

        return result
    }
}